web/modules/custom/joinup_invite/joinup_invite.module

```diff
@@ -2,5 +2,42 @@
 
 /**
  * @file
- * Contains joinup_invite.module.
+ * Hook implementations for the Joinup Invite module.
  */
+
+use Drupal\joinup_invite\Entity\InvitationInterface;
+
+/**
+ * Implements hook_ENTITY_TYPE_update().
+ */
+function joinup_invite_invitation_update(InvitationInterface $invitation) {
+  switch ($invitation->bundle()) {
+    case 'discussion':
+      // After an invitation to participate in a discussion has been accepted or
+      // rejected we should (un)subscribe the user and show them a success
+      // message.
```

Comment:
I am not fully satisfied with this approach. The PR introduces a new generic Invitation entity which can be used to invite users to whatever entity. The invitation is then accepted or rejected in the `InvitationController` but since this is also generic it doesn't know which action should be taken when an invitation is accepted or rejected.

I ended up hooking into `hook_ENTITY_TYPE_update()` and checking if the 'status' field has been changed to either `accepted` or `updated` but this approach seems hard to discover.

What approach would be the best?
* We could fire a dedicated `InvitationEvent` in the controller that already has the `accepted` or `rejected` status set.
* Another possible solution is to define `acceptInvitation()` and `rejectInvitation()` callbacks in the InvitationType configuration and call them directly.

Reply:
I also agree here. I was wondering the same thing. Since we have already the controller, why do we need to set the status messages here?

Reply:
For the same reason, the controller is generic for all kinds of invitations and doesn't know the right messages to display for each type of invitation. This should indeed also be solved.

Do you have a preference for either approach?

Reply:
Fixed it by implementing an event.